### PR TITLE
needed for custom field integration / data mapping

### PR DESCRIPTION
**Pull request by ottscho**
**1. Why is this change necessary?**
To use the custom fields in cms data mapping.

**2. What does this change do, exactly?**
Checks if is an array.

**3. Describe each step to reproduce the issue or behaviour.**
Overwrite cms mapping component and add custom fields to data mapping.
This changes needs to resolve custom fields array.

**4. Please link to the relevant issues (if any).**
5. https://issues.shopware.com/issues/NEXT-11407

### CODE REVIEW

File: src/Core/Content/Cms/DataResolver/Element/AbstractCmsElementResolver.php [hunk 1]
```diff
@@ -41,7 +41,11 @@ protected function resolveEntityValue(?Entity $entity, string $path)
             }
 
             try {
-                $value = $value->get($part);
+                if (is_array($value)) {
+                    $value = $value[$part];
+                } else {
+                    $value = $value->get($part);
+                }
```
Comment (mynameisbogdan) [inline src/Core/Content/Cms/DataResolver/Element/AbstractCmsElementResolver.php]: You can use the ternary operator here:
```suggestion
                $value = is_array($value) ? $value[$part] : $value->get($part);
```

Comment (NiklasLimberg) [reply src/Core/Content/Cms/DataResolver/Element/AbstractCmsElementResolver.php]: please implement that sugestion

Comment (mitelg) [reply src/Core/Content/Cms/DataResolver/Element/AbstractCmsElementResolver.php]: @NiklasLimberg is already...

**Comment by King-of-Babylon** [comment]
Thanks for your Contribution!

**Pull request closed without merging**